LibGit2Sharp/Core/ManagedHttpSmartSubtransport.cs

```diff
@@ -0,0 +1,278 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+
+namespace LibGit2Sharp.Core
+{
+    internal class ManagedHttpSmartSubtransport : RpcSmartSubtransport
+    {
+        protected override SmartSubtransportStream Action(string url, GitSmartSubtransportAction action)
+        {
+            string endpointUrl, contentType = null;
+            bool isPost = false;
+
+            switch (action)
+            {
+                case GitSmartSubtransportAction.UploadPackList:
+                    endpointUrl = string.Concat(url, "/info/refs?service=git-upload-pack");
+                    break;
+
+                case GitSmartSubtransportAction.UploadPack:
+                    endpointUrl = string.Concat(url, "/git-upload-pack");
+                    contentType = "application/x-git-upload-pack-request";
+                    isPost = true;
+                    break;
+
+                case GitSmartSubtransportAction.ReceivePackList:
+                    endpointUrl = string.Concat(url, "/info/refs?service=git-receive-pack");
+                    break;
+
+                case GitSmartSubtransportAction.ReceivePack:
+                    endpointUrl = string.Concat(url, "/git-receive-pack");
+                    contentType = "application/x-git-receive-pack-request";
+                    isPost = true;
+                    break;
+
+                default:
+                    throw new InvalidOperationException();
+            }
+
+            return new ManagedHttpSmartSubtransportStream(this, endpointUrl, isPost, contentType);
+        }
+
+        private class ManagedHttpSmartSubtransportStream : SmartSubtransportStream
+        {
+            private static int MAX_REDIRECTS = 7;
+
+            private MemoryStream postBuffer = new MemoryStream();
+            private HttpResponseMessage response;
+            private Stream responseStream;
+
+            private HttpClientHandler httpClientHandler;
+            private HttpClient httpClient;
+
+            public ManagedHttpSmartSubtransportStream(ManagedHttpSmartSubtransport parent, string endpointUrl, bool isPost, string contentType)
+                : base(parent)
+            {
+                EndpointUrl = new Uri(endpointUrl);
+                IsPost = isPost;
+                ContentType = contentType;
+
+                httpClientHandler = CreateClientHandler();
+                httpClient = new HttpClient(httpClientHandler);
```

Comment:
I wonder if turning off chunked encoding might fix the truncated response? 🤔 

```suggestion
                httpClient = new HttpClient(httpClientHandler);
                httpClient.DefaultRequestHeaders.TransferEncodingChunked = false;
```

We've been seeing a similar issue with GitHub Packages, which looks like it might besomehowe related to chunked encoding.

This is what happens when using `dotnet nuget push`:
```
error: An error occurred while sending the request.
error:   The response ended prematurely.
```

The same call without chunked encoding seems to works fine. 😕

Reply:
Amusingly, I just sorted out an issue with SecureTransport (the macOS TLS stack) and chunking.  https://github.com/libgit2/libgit2/pull/5432

Reply:
@jcansdale The request isn't being chunked. It's the response that is. And I don't think the client can control that. Setting the header as you propose would only claim the *request* doesn't carry a chunked payload I believe.